master/internal/stream/system_intg_test.go

```diff
@@ -18,36 +22,43 @@ import (
 	"github.com/determined-ai/determined/master/test/streamdata"
 )
 
+// simpleUpsert is for testing and just returns the preparable message that the streamer sends.
 func simpleUpsert(i stream.PreparableMessage) interface{} {
 	return i
 }
 
-func recordDeletion(s1, s2 string) interface{} {
-	fmt.Println("recording deletion:", s1+" and "+s2)
-	return s2
-}
-
+// startupReadWriter implements WebsocketLike and stores all messages received from streaming.
 type startupReadWriter struct {
-	data           []interface{}
-	startupMessage *StartupMsg
+	Data           []interface{}
+	StartupMessage *StartupMsg
+	Msg            interface{}
 }
 
+// ReadJSON sends the StartupMessage first, then send any Msg that is set.
 func (s *startupReadWriter) ReadJSON(data interface{}) error {
-	if s.startupMessage == nil {
-		return fmt.Errorf("startup message has been sent")
+	if s.StartupMessage != nil {
+		targetMsg, ok := data.(*StartupMsg)
+		if !ok {
+			return fmt.Errorf("target message type is not a pointer to StartupMsg")
+		}
+		targetMsg.Known = s.StartupMessage.Known
+		targetMsg.Subscribe = s.StartupMessage.Subscribe
+		s.StartupMessage = nil
+		return nil
 	}
-	targetMsg, ok := data.(*StartupMsg)
-	if !ok {
-		return fmt.Errorf("target message type is not a pointer to StartupMsg")
+	if s.Msg != nil {
+		_, ok := s.Msg.(*SubscriptionModMsg)
+		if !ok {
+			return fmt.Errorf("target message type is not a pointer to SubscriptionModMsg")
+		}
+		data = s.Msg
 	}
-	targetMsg.Known = s.startupMessage.Known
-	targetMsg.Subscribe = s.startupMessage.Subscribe
-	s.startupMessage = nil
 	return nil
 }
 
+// Write
```

Comment:
nit: I think you mean't to have more here 🤣

Reply:
😅 you're right! Will fill this out